hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/data/MeteredOutputStream.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.state.merkle.data;
+
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+public final class MeteredOutputStream extends FilterOutputStream {
```

Comment:
javadoc

Reply:
Also, if you weren't aware, something like this is already implemented in swirlds common, although this implementation is a bit simpler. For code examples, the unit test is the best place to start (if you are interested). https://github.com/swirlds/swirlds-platform/blob/develop/swirlds-unit-tests/common/swirlds-common-test/src/test/java/com/swirlds/common/test/io/extendable/CountingStreamExtensionTests.java